lib/foreman_maintain.rb

```diff
@@ -178,6 +178,13 @@ def perform_self_upgrade
       package_name, command = pkg_and_cmd_name
 
       puts "Checking for new version of #{package_name}..."
+
+      maintenance_module = 'satellite-maintenance:el8'
+      unless ForemanMaintain.package_manager.module_enabled?(maintenance_module)
+        puts "\nEnabling #{maintenance_module} module"
+        ForemanMaintain.package_manager.enable_module(maintenance_module)
+      end
```

Comment:
this is a common execution path, that is hit by all upgrades.
however, this module does not (and will never, with this name) exist upstream and might or might not exist downstream depending on the release.

additionally, I think similar enablement should happen in `definitions/scenarios/self_upgrade.rb` for the self-upgrade case (the code-path here is for the "upgrade me from the CURRENT release to allow upgrades to latest Z", not "upgrade me from the NEXT release to allow upgrades to NEXT")

Reply:
> > this is a common execution path, that is hit by all upgrades. however, this module does not (and will never, with this name) exist upstream and might or might not exist downstream depending on the release.
> 
> What are you suggesting here?

Something like `if module_exists?(module) && !module_enabled?(module) …`?

> 
> > upgrade case (the code-path here is for the "upgrade me from the CURRENT release to allow upgrades to latest Z", not "upgrade me from the NEXT release to allow upgrades to NEXT")
> 
> I slightly disagree with this, as this path is called anytime you run `upgrade check` or `upgrade run` which can happen for _any_ upgrade scenario.

But it is not called during `foreman-maintain self-upgrade`